### PR TITLE
fix: reloading docs viewer

### PR DESCRIPTION
**Pull request by acronymm**
### What does this PR do?
**TL;DR: Bug-fix: realoading document viewer.**

1. I added `{recipeName}` to the markup so that `renderedCallback()` gets called every time `recipeName` is updated. Though I'm not sure it is the best solution.
2. I updated `formatMarkdown()` function so that it replaces the existing content. (It used to just append the new content without removing the previous stuff)
3. I added `**/__mocks___/**` to `.forceignore`. Otherwise it gets deployed to the org via `sfdx force:source:deploy`.

### What issues does this PR fix or reference?

#[Issue #384](https://github.com/trailheadapps/apex-recipes/issues/384)

## The PR fulfills these requirements:

[v] Tests for the proposed changes have been added/updated.
[v] Code linting and formatting was performed.

### Functionality Before

Documentation tab doesn't reload and keeps showing docs of previous class.

### Functionality After

Documentation tab reloads and works as expected.

### CODE REVIEW

**Comment by pozil** [comment]
Closing as per my comments in #384
